File: packages/moss-repo-indexer-js/src/language.ts
import path from 'node:path'

const EXTENSION_LANGUAGE: Record<string, string> = {
  '.py': 'python',
  '.ts': 'typescript',
  '.tsx': 'typescript',
  '.js': 'javascript',
  '.jsx': 'javascript',
  '.go': 'go',
  '.rs': 'rust',
  '.java': 'java',
  '.md': 'markdown',
  '.mdx': 'markdown',
}

const MARKDOWN_EXTENSIONS = new Set(['.md', '.mdx'])

export function languageForPath(filePath: string): string {
  return EXTENSION_LANGUAGE[path.extname(filePath).toLowerCase()] ?? 'text'
}

export function isMarkdownPath(filePath: string): boolean {
  return MARKDOWN_EXTENSIONS.has(path.extname(filePath).toLowerCase())
}

File: packages/moss-repo-indexer/src/moss_repo_indexer/language.py
"""Map file extensions to language ids used in metadata."""

from __future__ import annotations

from pathlib import Path

_EXTENSION_LANGUAGE = {
    ".py": "python",
    ".ts": "typescript",
    ".tsx": "typescript",
    ".js": "javascript",
    ".jsx": "javascript",
    ".go": "go",
    ".rs": "rust",
    ".java": "java",
    ".md": "markdown",
    ".mdx": "markdown",
}

_MARKDOWN_EXTENSIONS = {".md", ".mdx"}


def language_for_path(path: Path) -> str:
    return _EXTENSION_LANGUAGE.get(path.suffix.lower(), "text")


def is_markdown_path(path: Path) -> bool:
    return path.suffix.lower() in _MARKDOWN_EXTENSIONS
